src/client/Client.cc

```diff
@@ -10679,7 +10679,9 @@ int Client::_do_setxattr(Inode *in, const char *name, const void *value,
   req->head.args.setxattr.flags = xattr_flags;
 
   bufferlist bl;
-  bl.append((const char*)value, size);
+  if (value != NULL) {
+    bl.append((const char*)value, size);
```

Comment:
For size = 0 it's fine, but for size > 0, bugs in external code will became invisible. Maybe it should assert() if value = NULL and size != 0?